react/features/filmstrip/components/native/TileView.js

```diff
@@ -0,0 +1,285 @@
+
+/**
+ * TileView component's property types.
+ */
+type Props = {
+
+    /**
+     * Whether or not the screen is currently in portrait orientation.
+     */
+    _isNarrowAspectRatio: boolean,
+
+    /**
+     * The participants in the conference.
+     */
+    _participants: Array<any>,
+
+    /**
+     * Invoked to update the receiver video quality.
+     */
+    dispatch: Dispatch<*>,
+
+    /**
+     * Callback to invoke when tile view is tapped.
+     */
+    onClick: Function
+};
+
+const TILE_ASPECT_RATIO = 1;
+
+/**
+ * Implements a React {@link Component} which represents the filmstrip on
+ * mobile/React Native.
+ *
+ * @extends Component
+ */
+class TileView extends Component<Props> {
+    /**
+     * Updates the receiver video quality.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this._updateReceiverQuality();
+    }
+
+    /**
+     * Updates the receiver video quality.
+     *
+     * @inheritdoc
+     */
+    componentDidUpdate() {
+        this._updateReceiverQuality();
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { onClick } = this.props;
+        const { height, width } = this._getScreenDimensions();
+        const rowElements = this._groupIntoRows(
+            this._renderThumbnails(), this._getColumnCount());
+
+        return (
+            <ScrollView
+                style = {{
+                    ...styles.tileView,
+                    height,
+                    width
+                }}>
+                <TouchableWithoutFeedback onPress = { onClick }>
+                    <View
+                        style = {{
+                            ...styles.tileViewRows,
+                            minHeight: height,
+                            minWidth: width
+                        }}>
+                        { rowElements }
+                    </View>
+                </TouchableWithoutFeedback>
+            </ScrollView>
+        );
+    }
+
+    /**
+     * Returns how many columns should be displayed for tile view.
+     *
+     * @returns {number}
+     * @private
+     */
+    _getColumnCount() {
+        const participantCount = this.props._participants.length;
+
+        // For narrow view, tiles should stack on top of each other for a lonely
+        // call and a 1:1 call. Otherwise tiles should be grouped into rows of
+        // two.
+        if (this.props._isNarrowAspectRatio) {
+            return participantCount >= 3 ? 2 : 1;
+        }
+
+        if (participantCount === 4) {
+            // In wide view, a four person call should display as a 2x2 grid.
+            return 2;
+        }
+
+        return participantCount >= 3 ? 3 : participantCount;
+    }
+
+    /**
+     * Returns the current height and width of the screen.
+     *
+     * @private
+     * @returns {Object}
+     */
+    _getScreenDimensions() {
```

Comment:
This won't work when multiple apps are used at once. Can you please use this to measure the available surface? https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/responsive-ui/components/DimensionsDetector.native.js